x509/verify.go

```diff
@@ -198,7 +209,7 @@ func (c *Certificate) buildChains(cache map[int][][]*Certificate, currentChain [
 		if err != nil {
 			continue
 		}
-		if !root.inChain(currentChain) {
+		if !root.subjectAndSPKIInChain(currentChain) {
```

Comment:
Thinking about this a bit more, I think `inChain` is probably correct. Consider a case where there's a "newer" version of a root that exists as both an intermediate with a parent of the old root, and a self-signed "new root". We'd never chain through the intermediate back to the old root.

Now, this might actually be invalid / not something that happens.